src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -88,38 +89,23 @@
   private final ElasticsearchSinkConnectorConfig config;
   private final ErrantRecordReporter reporter;
   private final RestHighLevelClient client;
-  private final ScheduledExecutorService executorService;
+  private final ExecutorService executorService;
   private final Time clock;
 
   public ElasticsearchClient(
       ElasticsearchSinkConnectorConfig config,
       ErrantRecordReporter reporter
   ) {
-    ConfigCallbackHandler configCallbackHandler = new ConfigCallbackHandler(config);
-    NHttpClientConnectionManager cm = configCallbackHandler.connectionManager();
-    /*
-     * Handles closing any idle or expired connections to avoid SocketTimeoutExceptions. Expired
-     * connections occur when the server closes their half of the connection without notifying
-     * the client.
-     */
-    this.executorService = Executors.newSingleThreadScheduledExecutor();
-    executorService.scheduleAtFixedRate(
```

Comment:
so where are the expired connections handled now?

Reply:
I wonder why we needed a schedued thread for this and shouldn't HTTP connection pool be doing it automatically?

Reply:
We don't explicitly call these methods anymore. With correctly configured request/socket timeouts, this should be handled automatically by the pool.

Reply:
could you please double-check why they were added in the first place by looking at git history?  Even I think HTTP client should handle idle/broken connections but I wonder how the pool will test, in MySQL connection pool we have to give a test query.

Reply:
I did, it was part of the initial Jest -> BulkProcessor migration, no extra information. 
What do you mean by "how the pool will test" ? Both methods refer to connections that are not in use.

Reply:
I mean we were calling closeExpiredConnections regularly but in the new code we aren't what does expiry has to do with correctly configured request/socket timeouts.

isn't expiry has to do with idle connections in pool that expired and now a new request uses that connection.

Reply:
I'm referring to the section where we say that we have this background tasks "to avoid SocketTimeoutExceptions". 
I meant that if timeouts are correctly configured, we should be getting these exceptions during the use of the connections, and we shouldn't have to deal with extra cleanups.

BTW, if a connections is expired, it should be leased by the pool.

Reply:
got it